src/components/ColorPickerContainer.ts

```diff
@@ -0,0 +1,286 @@
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+        this.subscriptionHandles = [];
+
+        if (mxObject) {
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                attr: this.props.colorAttribute,
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+        }
+    }
+
+    private handleSubscriptions = () => {
+        this.setState(this.validateColor(this.getValue(this.props.mxObject)));
+    }
+
+    private validateColor = (color: string): ColorPickerContainerState => {
+        let message = "";
+        if (color && this.props.format === "hex" && (color.indexOf("#") === -1)) {
+            message = `Color value ${color} should be of format '#d0d0d0'`;
+        } else if (color && this.props.format === "rgb" && (color.toLowerCase().indexOf("rgb(") === -1)) {
+            message = `Color value ${color} should be of format 'rgb(115,159,159)'`;
+        } else if (color && this.props.format === "rgba" && (color.toLowerCase().indexOf("rgba") === -1)) {
+            message = `Color value ${color} should be of format 'rgba(195,226,226,1)'`;
+        }
+
+        return {
+            alertMessage: message,
+            color,
+            displayColorPicker: this.state.displayColorPicker
+        };
+    }
+
+    private handleInputChange = (event: any) => {
+        const newColor = event.target.value as string;
+        if (newColor) {
+            this.colorChanged = this.getValue(this.props.mxObject) !== newColor;
+            const state = this.validateColor(event.target.value);
+            if (!state.alertMessage && this.props.mxObject) {
+                this.props.mxObject.set(this.props.colorAttribute, newColor);
+                this.handleOnChange();
+            }
+            this.setState(state);
+        } else {
+            this.setState({ alertMessage: "Invalid color", color: event.target.value });
+        }
+    }
+
+    private handleOnChange = () => {
+        const { mxform, mxObject, onChangeEvent, onChangeMicroflow, onChangeNanoflow, onChangePage } = this.props;
+        if (mxObject && ((this.previousColor !== this.state.color) || this.colorChanged)) {
+            const context = new mendix.lib.MxContext();
+            context.setContext(mxObject.getEntity(), mxObject.getGuid());
+            if (onChangeEvent === "callMicroflow" && onChangeMicroflow) {
+                window.mx.ui.action(onChangeMicroflow, {
+                    error: error => window.mx.ui.error(`Error while executing microflow ${onChangeMicroflow}: ${error.message}`), // tslint:disable-line max-line-length
+                    params: {
```

Comment:
use context, stead of params (shorter code) 
https://apidocs.mendix.com/7/client/mx.data.html#.action

Reply:
@Andries-Smit  Do we us `mx.data.action` or `mx.ui.action` when executing microflows?